torch/csrc/jit/script/compiler.cpp

```diff
@@ -542,33 +542,51 @@ static std::string prefixLine(const std::string& str, std::string prefix) {
 Value* emitBuiltinCall(
   const SourceRange& loc,
   Graph& graph,
-  Symbol name,
+  at::ArrayRef<Symbol> names,
   at::ArrayRef<NamedValue> inputs,
   at::ArrayRef<NamedValue> attributes,
   // if true, emitBuiltinCall will throw an exception if this builtin does not exist,
   // otherwise it will return nullptr if the builtin is not found.
   bool required) {
 
-  const auto& variants = getAllOperatorsFor(name);
+  size_t num_variants_found = 0;
   std::stringstream failure_messages;
-  for (const std::shared_ptr<Operator>& op : variants) {
-    if (auto result = tryEmitBuiltin(
-            op, failure_messages, loc, graph, name, inputs, attributes)) {
-      return result;
+  for (const auto& name : names) {
+    const auto& variants = getAllOperatorsFor(name);
+    num_variants_found += variants.size();
+
+    for (const std::shared_ptr<Operator>& op : variants) {
+      if (auto result = tryEmitBuiltin(
+              op, failure_messages, loc, graph, name, inputs, attributes)) {
+        return result;
+      }
     }
   }
   // none of the options worked
   if(!required) {
     return nullptr;
   }
-  if(variants.size() == 0) {
+  if(num_variants_found == 0) {
     throw ErrorReport(loc) << "unknown builtin op";
   }
   throw ErrorReport(loc) << "arguments for call are not valid:\n"
                          << prefixLine(failure_messages.str(), "  ")
                          << "for call at";
 }
 
+Value* emitBuiltinCall(
+  const SourceRange& loc,
+  Graph& graph,
+  Symbol name,
+  at::ArrayRef<NamedValue> inputs,
+  at::ArrayRef<NamedValue> attributes,
+  // if true, emitBuiltinCall will throw an exception if this builtin does not exist,
+  // otherwise it will return nullptr if the builtin is not found.
+  bool required) {
+  return emitBuiltinCall(
+      loc, graph, at::ArrayRef<Symbol>{name}, inputs, attributes, required);
```

Comment:
I'm pretty sure `ArrayRef` has an implicit constructor from a single element. Do we really need this function?

Reply:
oh nice TIL